src/Result/Api/ApiResultHandler.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: olsiqose
+ * Date: 09/07/2019
+ * Time: 11.51
+ */
+
+namespace Nosto\Result\Api;
+
+use Nosto\Result\ResultHandler;
+use Nosto\Request\Http\HttpResponse;
+
+abstract class ApiResultHandler extends ResultHandler
+{
+
+    protected function parseResponse(HttpResponse $response)
+    {
+        return $this->renderAPIResult($response);
+    }
+
+    /**
+     * @param HttpResponse $response
+     * @return string|array
+     */
+    abstract protected function renderAPIResult(HttpResponse $response);
```

Comment:
Why are there two methods called `parseResponse` and `renderAPIResult`?
Why is it called "render" if it is returning something? Render implies that it would render something and return.